src/services/services.ts

```diff
@@ -2797,18 +2800,30 @@ namespace ts {
             return program.getOptionsDiagnostics().concat(program.getGlobalDiagnostics());
         }
 
-        /// Completion
-        function getCompletionEntryDisplayNameForSymbol(symbol: Symbol, target: ScriptTarget, performCharacterChecks: boolean): string {
-            let displayName = symbol.getName();
-            if (displayName) {
-                // If this is the default export, get the name of the declaration if it exists
-                if (displayName === "default") {
-                    let localSymbol = getLocalSymbolForExportDefault(symbol);
-                    if (localSymbol && localSymbol.name) {
-                        displayName = symbol.valueDeclaration.localSymbol.name;
-                    }
-                }
-
+        /**
+         * Get the name to be display in completion from a given symbol.
+         *
+         * @return undefined if the name is of external module otherwise a name with striped of any quote
+         */
+        function getCompletionEntryDisplayNameForSymbol(symbol: Symbol, target: ScriptTarget, performCharacterChecks: boolean, location: Node): string {

```

Comment:
As per our discussion, there is another way to fix the case of classExpression and functionExpression which is to change bindAnonymouseDeclaration to bind classExpression or functionExpression to its declared name if it has one. I will experiment with that in a separate PR